apps/chat2/config_chat2.nim

```diff
@@ -290,6 +290,11 @@ type
       desc: "Address of membership contract on an Ethereum testnet", 
       defaultValue: ""
       name: "rln-relay-eth-contract-address" }: string
+
+    rlnRelayCredentialsPassword* {.
+      desc: "Password for encrypting RLN credentials", 
+      defaultValue: ""
+      name: "rln-relay-cred-password" }: string
```

Comment:
Can this configuration option be within a `when defined(rln):` compilation guard?

The rest of the RLN conf options should also be under the `when` compilation guard; if this is allowed by Nim.

Reply:
I'm ok moving this to another PR. Not a blocker on this PR, but certainly, in the near future, we should put a `when` guard to RLN-specific configuration options (the same way we have it in the implementation).

Reply:
One more relevant issue about the visibility of config options is https://github.com/status-im/nwaku/issues/999 where it was suggested to use sub-commands to group all the rln-relay-related configs and make them available only when the rln-relay config option is set to true. Would that address your comment @LNSD ?

Reply:
Currently, we have the `RLN=true` feature flag (which translates into `-d:rln` and `-d:rlnzerokit`). But this compilation feature flag is not being applied in the configuration. 

If you compile with `RLN=false`, you still see the RLN configuration options in the help. That can be misleading because no RLN functionality was compiled.

So the solution would be to wrap in a compilation guard all the RLN options:

```Nim
    when defined(rln) or defined(rlnzerokit):
      # [...]
      
      rlnRelayCredentialsPassword* {.
        desc: "Password for encrypting RLN credentials", 
        defaultValue: ""
        name: "rln-relay-cred-password" }: string

      # [...]
```

---- 

On a side note, I don't understand how subcommands will help here. We are not using subcommands in the nwaku node app (aka `wakunode2`). And I don't see any use case for them in `wakunode2`, at least in the near future. 

We are working on grouping/namespacing the configuration options with a prefix (e.g., `--store-message-db-url`, `--store-message-retention-policy`, etc.). But this is already present in the current implementation of RLN 😁

Reply:
>  but nim seems a bit unhandy for these tasks.

Now I see why you said that. It is an issue in `nim-confutils` macros 😫 Then, let's leave it as it is until we have a better way of doing this.

Reply:
@LNSD I see your point, that makes sense. 

> On a side note, I don't understand how subcommands will help here. We are not using subcommands in the nwaku node app (aka wakunode2). And I don't see any use case for them in wakunode2, at least in the near future.

It was suggested in one of the PRs and my understanding was that it is part of the plan to use subcommands: https://github.com/status-im/nwaku/pull/992#discussion_r893633559 @jm-clius

Reply:
Please let us know if the plan re subcommands has changed so that we deprioritize that issue i.e., #999 cc: @jm-clius @LNSD

Reply:
> I don't understand how subcommands will help here.

I'd say, then all the rln-relay-related config options will get encapsulated (and only accessible) under the rln-relay command/subcommand, hence the CLI interface becomes less confusing for the users (I assume this is the desirable feature we are looking for)